src/main/scala/lenthall/validation/ErrorOr.scala

```diff
@@ -19,4 +21,10 @@ object ErrorOr {
       }
     }
   }
+
+  implicit class MapErrorOrRhs[A, B](val m: Map[A, ErrorOr[B]]) extends AnyVal {
+    private def elementToErrorOr(kvp: (A, ErrorOr[B])): ErrorOr[(A, B)] = kvp._2 map { v => (kvp._1, v) }
+
+    def sequence: ErrorOr[Map[A, B]] = m.toList.traverse[ErrorOr, (A, B)] { x => elementToErrorOr(x) } map { _.toMap }
```

Comment:
ToL could you implement `traverse` and have `sequence` be a special case that applies `identity` to the elements?